packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -50,20 +50,36 @@ addFilter(
 export default function CollabSidebar() {
 	const { createNotice } = useDispatch( noticesStore );
 	const { saveEntityRecord, deleteEntityRecord } = useDispatch( coreStore );
-	const { getEntityRecords, getEntityRecord } = resolveSelect( coreStore );
+	const { getEntityRecord } = resolveSelect( coreStore );
 
 	// eslint-disable-next-line @wordpress/data-no-store-string-literals
 	const { enableComplementaryArea } = useDispatch( interfaceStore );
 	const [ blockCommentID, setBlockCommentID ] = useState( null );
 	const [ showCommentBoard, setShowCommentBoard ] = useState( false );
 
-	const [ threads, setThreads ] = useState( () => [] );
 	const { postId } = useSelect( ( select ) => {
 		return {
 			postId: select( editorStore ).getCurrentPostId(),
 		};
 	}, [] );
 
+	const threads = useSelect(
+		( select ) => {
+			if ( ! postId ) {
+				return [];
+			}
+			const { getEntityRecords } = select( coreStore );
+			const data = getEntityRecords( 'root', 'comment', {
+				post: postId,
+				type: 'block_comment',
+				status: 'any',
+				per_page: 100,
+			} );
+			return Array.isArray( data ) ? data : [];
```

Comment:
We shouldn't create a new array inside `useSelect`, it will create excessive re-renders. `useSelect` will think there's change. Can you create an EMTPY_ARRY constant outside this component. Same for the early return above.

Reply:
Alternatively, let it return null and move the array checks outside of `useSelect`.